docs/source/accelerators/hpu.rst

```diff
@@ -0,0 +1,133 @@
+
+.. code-block:: python
+
+    trainer = Trainer(devices=8, accelerator="hpu")
+
+.. note::
+    If the ``devices`` flag is not defined, it will assume ``devices`` to be ``"auto"`` and fetch the :meth:`~pytorch_lightning.accelerators.hpu.HPUAccelerator.auto_device_count`
+    from :class:`~pytorch_lightning.accelerators.hpu.HPUAccelerator`.
+
+
+Mixed Precision Plugin
+----------------------
+
+Lightning also allows mixed precision training with HPUs.
+By default, HPU training will use 32-bit precision. To enable mixed precision, set the ``precision`` flag.
+
+In addition to the default settings in HMP, users also have the option of overriding these defaults and providing their own BF16 and FP32 operator lists using the ``plugins`` parameter of Trainer class.
+HPU's precision plugin is realised using :class:`~pytorch_lightning.plugins.precision.hpu.HPUPrecisionPlugin`. The ``hmp_params`` parameter with this plugin is used to override the default operator list. An example can be found in the subsequent section.
+
+.. code-block:: python
+
+    trainer = Trainer(devices=1, accelerator="hpu", precision=16)
+
+
+Enabling Mixed Precision Options
+--------------------------------
+
+Internally, :class:`~pytorch_lightning.plugins.precision.HPUPrecisionPlugin` uses the Habana Mixed Precision (HMP) package to enable mixed precision training.
+
+You can execute the ops in FP32 or BF16 precision. The HMP package modifies the python operators to add the appropriate cast operations for the arguments before execution.
+The default settings enable users to enable mixed precision training with minimal code easily.
+
+In addition to the default settings in HMP, users also have the option of overriding these defaults and providing their BF16 and FP32 operator lists by passing it
+to the ``hmp_params`` parameter of :class:`~pytorch_lightning.plugins.precision.HPUPrecisionPlugin`.
+
+The below snippet shows an example model using MNIST with a single Habana Gaudi device and making use of HMP by overriding the default parameters.
+This enables advanced users to provide their own BF16 and FP32 operator list instead of using the HMP defaults.
+
+.. code-block:: python
+
+    import pytorch_lightning as pl
+    from pytorch_lightning.plugins import HPUPrecisionPlugin
+
+
+    class LitClassifier(pl.LightningModule):
+        def __init__(self):
+            super(LitClassifier, self).__init__()
+
+        ...
+
+
+    # Init our model
+    model = LitClassifier()
+
+    # Init DataLoader from MNIST Dataset
+    dm = MNISTDataModule(batch_size=batch_size)
+
+    ...
+
+    # Optional Habana mixed precision params to be set
+    hmp_keys = ["level", "verbose", "bf16_ops", "fp32_ops"]
+    hmp_params = dict.fromkeys(hmp_keys)
+    hmp_params["level"] = "O1"
+    hmp_params["verbose"] = False
+    hmp_params["bf16_ops"] = "ops_bf16_mnist.txt"
+    hmp_params["fp32_ops"] = "ops_fp32_mnist.txt"
```

Comment:
Why this pattern instead of creating the dicitoanry directly?

```suggestion
    hmp_params = {
        'level': 'O1',
        'verbose': False,
        'bf16_ops': 'ops_bf16_mnist.txt',
        'fp32_ops': 'ops_fp32_mnist.txt'
    }
```